plugins/scaffolder-backend/src/scaffolder/stages/publish/github.ts

```diff
@@ -28,8 +31,13 @@ export class GithubPublisher implements PublisherBase {
     config: GitHubIntegrationConfig,
     { repoVisibility }: { repoVisibility: RepoVisibilityOptions },
   ) {
+    let credentialsProvider: GithubCredentialsProvider | undefined = undefined;
+
     if (!config.token) {
-      return undefined;
+      if (!config.apps) {
+        return undefined;
+      }
+      credentialsProvider = GithubCredentialsProvider.create(config);
```

Comment:
You can instantiate the `GithubCredentialsProvider` unconditionally, it'll take care of figuring out if tokens should be grabbed from app, static token, or if no token is available

Reply:
instantiate of the `GithubCredentialsProvider` is now unconditional.